crypto/src/main/scala/org/bitcoins/crypto/AdaptorUtil.scala

```diff
@@ -0,0 +1,133 @@
+package org.bitcoins.crypto
+
+import scodec.bits.ByteVector
+
+object AdaptorUtil {
+  import ECAdaptorSignature.{deserializePoint, serializePoint}
+
+  // Compute s' = k^-1 * (dataToSign + rx*privateKey)
+  private def adaptorSignHelper(
+      dataToSign: ByteVector,
+      k: FieldElement,
+      r: ECPublicKey,
+      privateKey: ECPrivateKey): FieldElement = {
+    CryptoUtil.decodePoint(r) match {
+      case ECPointInfinity =>
+        throw new IllegalArgumentException("Invalid point")
```

Comment:
nit: 

```scala
throw new IllegalArgumentException(s"Invalid point, got ${ECPointInfinity}")
```

Generally putting the input in the error message is _really_ helpful when debugging.